http/write_handler.go

```diff
@@ -306,6 +275,38 @@ func decodeWriteRequest(ctx context.Context, r *http.Request) (*postWriteRequest
 	}, nil
 }
 
+func readWriteRequest(ctx context.Context, rc io.ReadCloser, encoding string, maxBatchSizeBytes int64) (v []byte, err error) {
+	defer func() {
+		// close the reader now that all bytes have been consumed
+		// this will return non-nil in the case of a configured limit
+		// being exceeded
+		if cerr := rc.Close(); err == nil {
+			err = cerr
+		}
+	}()
+
+	if encoding == "gzip" {
```

Comment:
also check if it is `x-gzip`, as http 1.1 spec recommends you check for it as well, as an alias for `gzip`